components/aws/sagemaker/common/_utils.py

```diff
@@ -471,10 +471,52 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  try:
+      endpoint_name = client.describe_endpoint(EndpointName=endpoint_name)['EndpointName']
+      logging.info("Endpoint exists: " + endpoint_name)
+      return True
+  except ClientError as e:
```

Comment:
for these` _exists` methods I think we should check for specific error before concluding resource does not exist and raise Exceptions for other cases

something like
```
try:
   <code>
catch ValidationException:
   log("endpoint does not exist")
   pass
catch otherExceptions:
   raise Exception(failed to describe for other reason, e.Error.Message)
```

Reply:
Since there is no specific exception to catch for Endpoint not found, it is not going to help if we caught the ValidationException and it can be triggered in other cases too